### PR TITLE
Add debug token caching to app-check-exp

### PR DESCRIPTION
**Pull request by hsubox76**
Ported from v8: https://github.com/firebase/firebase-js-sdk/pull/5055
Also includes some of the test reliability fixes in 5055 that I missed porting over in an earlier PR.

### CODE REVIEW

File: packages-exp/app-check-exp/src/internal-api.test.ts [hunk 1]
```diff
@@ -239,6 +236,7 @@ describe('internal api', () => {
     });
 
     it('ignores listeners that throw', async () => {
+      stub(console, 'error');
```
Comment (Feiyang1) [inline packages-exp/app-check-exp/src/internal-api.test.ts:239]: Is it to just suppress error messages during testing? If so, can you please add comments to clarify?

File: packages-exp/app-check-exp/src/internal-api.ts [hunk 2]
```diff
@@ -152,50 +161,31 @@ export function addTokenListener(
...
-      !newState.tokenRefresher.isRunning() &&
-      state.isTokenAutoRefreshEnabled === true
-    ) {
-      newState.tokenRefresher.start();
-    }
+  // Create the refresher but don't start it if `isTokenAutoRefreshEnabled`
+  // is not true.
+  if (
+    !newState.tokenRefresher.isRunning() &&
+    state.isTokenAutoRefreshEnabled === true
```
Comment (Feiyang1) [inline packages-exp/app-check-exp/src/internal-api.ts:176]:
```suggestion
    state.isTokenAutoRefreshEnabled
```
We generally don't use `===` explicitly for booleans.

File: packages-exp/app-check-exp/src/internal-api.test.ts (hunk 1 above)
Comment (hsubox76) [reply packages-exp/app-check-exp/src/internal-api.test.ts:239]: Yes, done.

File: packages-exp/app-check-exp/src/internal-api.ts (hunk 2 above)
Comment (hsubox76) [reply packages-exp/app-check-exp/src/internal-api.ts:176]: Fixed.

Comment (Feiyang1) [reply packages-exp/app-check-exp/src/internal-api.ts:176]: It doesn't appear to be fixed.

**Review by Feiyang1 — APPROVED** [review]

**Pull request merged**